airbyte-ci/connectors/metadata_service/lib/metadata_service/gcs_upload.py

```diff
@@ -176,6 +189,58 @@ def _doc_upload(
     return doc_uploaded, doc_blob_id
 
 
+def _file_upload(
```

Comment:
I wonder if it would be easier to have `_file_upload` upload once, with a tag that can be either a string version, or latest. That could return a named `@dataclass` instead of a tuple just for readability. And we could call this for both latest and version number when required.

This is purely stylistic / readability, so feel free to ignore me on this.

Reply:
I appreciate this feedback but without larger refactoring effort, I think this is the most streamlined option we have. I started with the approach you recommend, actually, but because we redundantly have to pass the bucket and several other args, it was much more verbose in that form - and then the "if/then" logic was also having to be repeated in several places.

One thing I can do to improve readability without a lot of refactoring would be to use the Named tuple or type alias approach that @bnchrch suggested in another thread. 

https://github.com/airbytehq/airbyte/pull/42584#discussion_r1697339609

@bnchrch suggests "MaybeUpload" as a pairing of the bool/str pair, which I think simplifies the return signature significantly. We could do other refactoring - such as using the null string as a replacement for a falsey bool - but I'm reluctant to do such a large refactor here where we're just trying to add more files to the process.